### PR TITLE
[doc] Add contribution header pages

### PR DESCRIPTION
**Pull request by marnovandermaas**
I added header pages for the pages in the contribution chapter. I also fleshed out the cores page a bit more, since it seemed a bit bare.

One nice thing about adding the header pages is that I think it looks visually better in the book.

Before:
image

After:
image

### CODE REVIEW

**Review by jwnrt — APPROVED** [review]
Thanks, this looks a lot nicer. LGTM.

**Review by HU90m — APPROVED** [review]
This makes a world of difference to the feel of the documentation. Thanks for filling out the draft chapters.

File: doc/contributing/hw/README.md [hunk 1]
```diff
@@ -0,0 +1,10 @@
+# Contributing to Hardware
+
+OpenTitan is a silicon project, so naturally there is a lot of aspects of hardware to contribute to.
```
Comment (hcallahan-lowrisc) [inline doc/contributing/hw/README.md:3]:
```suggestion
OpenTitan is an open-source silicon project, so naturally there are lots of aspects of the hardware design to contribute to.
```

**Review by hcallahan-lowrisc — APPROVED** [review]
Great! Thanks for the changes @marnovandermaas that keep increasing our polish like this. Just some suggestions on a few bits of prose.

File: hw/doc/cores.md [hunk 2]
```diff
@@ -1,4 +1,8 @@
 # Cores
 
+Cores in OpenTitan are processing units that can run programs.
+Currently there are the following cores in OpenTitan:
 * [Ibex](../ip/rv_core_ibex/README.md)
 * [OTBN](../ip/otbn/README.md)
```
[the remark this answers is not in the document]
Comment (marnovandermaas) [reply hw/doc/cores.md]: Thanks for suggesting these changes, I have added them to the PR.

File: doc/contributing/sw/README.md [hunk 3]
```diff
@@ -0,0 +1,7 @@
+# Contributing to Software
+
+Software is an important part of testing and deploying OpenTitan.
+It is good to realize that there are multiple [cores](../../../hw/doc/cores.md) in OpenTitan that can run software.
```
[the remark this answers is not in the document]
Comment (marnovandermaas) [reply doc/contributing/sw/README.md]: Thanks for suggesting these changes, I have added them to the PR.

File: doc/contributing/dv/README.md [hunk 4]
```diff
@@ -0,0 +1,7 @@
+# Contributing to Verification
+
+Verification is an essential part of making OpenTitan a system that can be used throughout the industry.
```
[the remark this answers is not in the document]
Comment (marnovandermaas) [reply doc/contributing/dv/README.md]: Thanks for suggesting these changes, I have added them to the PR.

File: doc/contributing/hw/README.md (hunk 1 above)
Comment (marnovandermaas) [reply doc/contributing/hw/README.md]: Thanks for suggesting these changes, I have added them to the PR.

**Review by engdoreis — APPROVED** [review]

**Comment by marnovandermaas** [comment]
Rebase to re-trigger CI.

**Pull request merged**